contracts/contribute/ContributeManager.sol

```diff
@@ -352,4 +508,25 @@ contract ContributeManager is ERC721TokenReceiver {
 
         _locked = 1;
     }
+
+    /// @dev Pulls unbonded service by contributor.
+    function pullUnbondedService() external {
```

Comment:
Self-explaining: after the service is unstaked, terminated and unbonded: choose to pull back to contributor.